lua/weapons/weapon_ttt_m16_stn.lua

```diff
@@ -0,0 +1,158 @@
+-- NLL created by https://steamcommunity.com/profiles/76561198813867340
+
+include("autorun/nl_library.lua")
+
+if SERVER then
+    AddCSLuaFile()
+end
+
+DEFINE_BASECLASS("weapon_tttbase")
```

Comment:
By the way, if you want this to work **exactly** like the M16 just with a tweaks to how it fires, you can use the M16 SWEP as the baseclass. Then you don't need to copy/rewrite everything from the M16, you can just overwrite what you need to.